src/internal/storage/Storage.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <concepts>
+#include <functional>
+#include <string>
+
+#include <external/expected/Expected.h>
+
+namespace internal::storage {
+
+class Position {
+public:
+	Position() = default;
+	Position(const Position &) = default;
+
+	/*
+	 * Intentionally implicit
+	 */
+	Position(long pos) : m_pos{pos}, m_isset{true} {}
+	operator long() { return m_pos; }
```

Comment:
Can be made const and should be marked explicit

Reply:
As far as I am aware there should not be `const`. Check [this](https://stackoverflow.com/questions/8406898/benefits-of-using-const-with-scalar-type-e-g-const-double-or-const-int) out.